src/Diagnostics.DataProviders/DataProviders/Base/SupportObserverDataProviderBase.cs

```diff
@@ -25,45 +26,61 @@ public SupportObserverDataProviderBase(OperationDataCache cache, SupportObserver
             FillObserverRouteTemplates();
         }
 
-        public async Task<dynamic> GetResource(string observerUrl)
+        public Task<dynamic> GetResource(string resourceUrl)
         {
             Uri uri;
-            Dictionary<string, string> routeParametersAndValues = null;
-            string routeTemplate = null;
+
+            var allowedHosts = new string[] { "wawsobserver.azurewebsites.windows.net", "support-bay-api.azurewebsites.net", "support-bay-api-stage.azurewebsites.net" };
```

Comment:
In case someone wants a new API, i can first add it to staging then I will tell them to experiment writing their detector against that and if all looks good I will deploy to production.